src/components/Icon/Icon.tsx

```diff
@@ -213,15 +215,35 @@ function Icon({
     backdrop && styles.hasBackdrop,
   );
 
+  const defaultIconProps = {
+    className: styles.Svg,
+    focusable: 'false',
+    'aria-hidden': 'true',
+  };
+
+  let SourceComponent = source as React.SFC<React.SVGProps<SVGSVGElement>>;
   let contentMarkup: React.ReactNode;
   if (source === 'placeholder') {
     contentMarkup = <div className={styles.Placeholder} />;
+  } else if (isBundledIcon(source)) {
+    SourceComponent = BUNDLED_ICONS[source];
+    contentMarkup = <SourceComponent {...defaultIconProps} />;
+  } else if (
+    typeof source === 'function' &&
+    React.isValidElement(<SourceComponent />)
```

Comment:
By repeating the <SourceComponent> here in the if clause and again in the if body I think we end up invoking the component twice - can get that down to one by tweaking the ordering of this somehow, or assuming that if `typeof source === function` then it's going to be a React SFC?

Reply:
I thought about this too. I do want to keep the `React.isValidElement` check in there though.

Right now, the Icon component just fails silently if you pass in something that it doesn't understand. I want to change that in a follow-up PR and make it throw an exception, but without the `isValidElement` check in there this would accept any function and not be as useful.

What do you think of something like this:
```js
} else if (typeof source ==='function') {
  const functionOutput = (<SourceComponent {...defaultIconProps} />);
  if (React.isValidElement(functionOutput)) {
    contentMarkup = functionOutput;
  }
}
```